### PR TITLE
[Dashboard] feat: reputation chart dropdown menu

### LINKED ISSUE #3060: [Dashboard] Reputation card kebab menu
### Story

- As a USER, I would like to be able to see a breakdown of the percentage influence in reputation of the top teams in the organisation.

### Description

We want to display a kebab menu on the Reputation card, having the following options

- View all teams
    - Should navigate the user to the Teams page.
- Create new team
    - Should open a new “Create new team” action in the action panel.
- Edit a team
    - Should open a new “Edit team” action in the action panel.
- Deprecate a team
    - Should open a new “Archive a team” action in the action panel.

Image

### This issue is complete when...

- [ ] Dashboard page Reputation card displays the menu icon
- [ ] Clicking on the menu icon opens the menu having the above mentioned options
- [ ] Clicking on `View all teams` redirects to Teams page
- [ ] Clicking on `Create new team` opens the Action Panel with `Create new team` action selected
- [ ] Clicking on `Edit a team` opens the Action Panel with `Edit a team` action selected
- [ ] Clicking on `Deprecate a team` opens the Action Panel with `Archive a team` action selected - this can be hidden for the moment

[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4867-41982&node-type=FRAME&t=uhZRrsioRrM0wDbQ-0)

**Feature branch**: `feat/enhanced-dashboard`

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Implement dropdown chart with team actions for the reputation donut chart.
I didn't use our existing `DropdownMenu` component, because it's way too cumbersome and I've instead opted for a simple solution using `popper` and a `div`. 
I think we will be able to decouple this in due time, but I didn't start on it right now.

[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4878-88566&t=ZEsiLeswZqxoEGLv-4)
image

### Functionality

1. Verify that the dropdown opens showing all 3 items
image
2. Clicking on `View all teams` goes to the teams page
3. Clicking on `Create new team` opens the sidebar with the `Create new team` option preselected
4. Clicking on `Edit new team` opens the sidebar with the `Edit new team` option preselected

### UI

I couldn't find any designs for the hover state so I just tried to replicate what we have on our current dropdowns :shrug:

1. Verify that the three dots turn blue when the menu is active and on hover
image
2. Verify that the items turn bold on hover and have a rounded gray background 
image
3. Verify that on mobile the dropdown is full width and has padding left and right
image

## Diffs

**New stuff** ✨

* `TeamActionsMenu` component

Resolves #3060

### CODE REVIEW

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  editTeam: {
+    id: `${displayName}.editTeam`,
+    defaultMessage: 'Edit team',
+  },
+});
+
+// @TODO maybe decouple some of this into a dropdown menu component?
+// eslint-disable-next-line max-len
+const dropdownItemClassName = tw`flex w-full items-center gap-2 rounded-s px-3 py-2 text-gray-900 hover:font-medium md:hover:bg-gray-50`;
```
Comment (mmioana) [inline src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx:35]: Here we should add also `text-md`

Current implementation
Screenshot 2024-09-17 at 13 30 26

Figma
Screenshot 2024-09-17 at 13 33 46

**Review by mmioana — CHANGES_REQUESTED** [review]
Awesome and simple solution @bassgeta 🎉 

Screenshot 2024-09-17 at 13 37 19
Screenshot 2024-09-17 at 13 31 25

Clicking the items navigate to the correct page or open the correct action

https://github.com/user-attachments/assets/7dd4e4f8-7765-44b8-850c-313254fe6f8f

There is a font-size mismatch compared to Figma for which I left a comment, if you can please tackle it 🙏

**Review by mmioana — APPROVED** [review]
Thank you for the change @bassgeta 💯 now it looks perfect

Screenshot 2024-09-18 at 11 58 11

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            <Link to={`/${colonyName}/teams`} className={dropdownItemClassName}>
+              <Eye size={18} />
+              <span>{formatText(MSG.viewAllTeams)}</span>
+            </Link>
+            <button
+              type="button"
+              className={dropdownItemClassName}
+              onClick={handleCreateTeamClick}
+            >
+              <Lock size={18} />
```
Comment (rumzledz) [inline src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx:99]: This needs to use the `LockKey` icon based on Figma:

```js
<LockKey size={18} />
```

Screenshot 2024-09-18 at 19 59 35

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    setIsDropdownOpen(false);
+  };
+
+  return (
+    <>
+      <button
+        type="button"
+        ref={setTriggerRef}
+        className={clsx('hover:text-blue-400', {
+          'text-gray-600': !visible,
```
Comment (rumzledz) [inline src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/TeamActionsMenu.tsx:77]: This needs to be color `gray-400` in its non-hovered/non-active state based on the [Figma designs](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4878-76054&t=27Zn8LxD7w4sTlH7-4)

```js
className={clsx('text-gray-400 hover:text-blue-400', {
  '!text-blue-400': visible,
})}
```

image

**Review by rumzledz — CHANGES_REQUESTED** [review]
Great work on this one @bassgeta ! 🔥 I can confirm that the buttons are doing the right thing:



I left some really minor UI comments but that's pretty much it!

**Review by rumzledz — APPROVED** [review]
Thanks for the updates @bassgeta, LGTM! 🚀

**Review by davecreaser — APPROVED** [review]
Nice one @bassgeta, works like a charm and code looks good and clean :)

Tested all the links funcitonalities, hover states, and mobile as you suggested:

Screenshot 2024-09-19 at 16 03 21
Screenshot 2024-09-19 at 16 04 46

**Pull request merged**